code/src/components/Labyrinth.js

```diff
@@ -0,0 +1,122 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { nextStep, game } from "../reducers/game";
+
+// data is collected from the slice (game) with currentposition from the initialstate.
+export const Labyrinth = () => {
+  const { description, coordinates, actions } = useSelector(
+    (store) => store.game.currentPosition
+  );
+  const dispatch = useDispatch();
+
+  const handleButtonClick = (type, direction) => {
+    dispatch(game.actions.setPreviousCoordinates(coordinates));
+    // call the api, pass along type and direction to use as part of the body
+    dispatch(nextStep(type, direction));
+  };
+
+  // changed the logic of direction to the opposite, to match our css grid
+  const ActionCard = ({ description, type, direction }) => {
```

Comment:
So curious and trying to understand why the logic works like this (and you are forced to add this function) 🤔